### PR TITLE
Show proper error message when migrating PJ without frameworks section.

### PR DESCRIPTION
**Pull request by livarcocc**
When we had a project.json without frameworks, we were failing to show the appropriate error message because when we were generating the error message itself we tried to access the project context (which requires a framework in the PJ). This fix makes the ProjectContext optional to add the migration error.

Fixes https://github.com/dotnet/cli/issues/4570

@piotrpMSFT @krwq @jgoshi

### CODE REVIEW

File: TestAssets/TestProjects/TestLibraryWithProjectFileWithoutFrameworks/project.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "dependencies": {
+        "Microsoft.VisualStudio.Web.CodeGeneration.Tools" : {
+            "version": "1.0.0-preview2-final",
+            "type": "build"
+        }
+    }
+}
```
Comment (piotrpMSFT) [inline TestAssets/TestProjects/TestLibraryWithProjectFileWithoutFrameworks/project.json]: newline

Comment (piotrpMSFT) [inline TestAssets/TestProjects/TestLibraryWithProjectFileWithoutFrameworks/project.json]: This likely needs to go in nonrestoredprojects. Does restore work without frameworks?

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs [hunk 2]
```diff
@@ -77,6 +77,32 @@ public void It_has_error_when_migrating_a_non_csharp_app()
             errorMessage.Should().Contain("MIGRATE20013::Non-Csharp App: Cannot migrate project");
         }
 
+        [Fact]
+        public void ItHasErrorWhenMigratingAProjectJsonWithoutAFrameworks()
+        {
+            var testAppName = "TestLibraryWithProjectFileWithoutFrameworks";
+            var testInstance = TestAssets.Get(testAppName)
+                .CreateInstance(testAppName)
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs]: Why is this overriding the default name? two tests can clash.

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs [hunk 3]
```diff
@@ -77,6 +77,32 @@ public void It_has_error_when_migrating_a_non_csharp_app()
             errorMessage.Should().Contain("MIGRATE20013::Non-Csharp App: Cannot migrate project");
         }
 
+        [Fact]
+        public void ItHasErrorWhenMigratingAProjectJsonWithoutAFrameworks()
+        {
+            var testAppName = "TestLibraryWithProjectFileWithoutFrameworks";
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs]: I don't think we need a var here, I'd just inline the name so we don't need to reason about it in the test code.

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs [hunk 4]
```diff
@@ -77,6 +77,32 @@ public void It_has_error_when_migrating_a_non_csharp_app()
...
+                mockProj);
+
+            var projectMigrator = new ProjectMigrator(new FakeEmptyMigrationRule());
+            var report = projectMigrator.Migrate(testSettings);
+
+            var projectReport = report.ProjectMigrationReports.First();
+
+            var errorMessage = projectReport.Errors.First().GetFormattedErrorMessage();
+            errorMessage.Should().Contain("MIGRATE1013::No Project:");
+            errorMessage.Should().Contain($"The project.json specifies no target frameworks in {testProjectDirectory}");
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs]:
```
report.ProjectMigrationReports.First()
  .Errors.First()
  .GetFormattedErrorMessage()
  .Should.Contain(...)
      .And.Contain(...);      
```

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs (hunk 3 above)
Comment (livarcocc) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs]: Fixed.

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs (hunk 2 above)
Comment (livarcocc) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs]: Fixed.

**Comment by livarcocc** [comment]
@MattGertz

**Comment by livarcocc** [comment]
@dotnet-bot Test RHEL7.2 x64 Release Build

**Pull request merged**